todo_app/app.py

```diff
@@ -1,46 +1,76 @@
 from flask import Flask, redirect, render_template, request
-from todo_app.data.session_items import get_items, get_item, add_item, save_item, delete_item
+from datetime import datetime
+
+from todo_app.data.trello_items import Trello
+from todo_app.utils import get_trello_credentials
 
 from todo_app.flask_config import Config
 
-app = Flask(__name__)
+app = Flask(__name__, static_folder='static', static_url_path='')
 app.config.from_object(Config())
 
 
 @app.route('/')
 def index():
-    todo_items = get_items()
+
+    trello = Trello()
+
+    todo_items = trello.get_items_sorted()
     num_todos = len(todo_items)
-    num_complete_todos = len([item for item in todo_items if item['status'] == 'Completed'])
-    sorted_items = sorted(todo_items, key = lambda item: item['status'], reverse=True)
+    num_complete_todos = len([item for item in todo_items if item.status == 'Completed'])
+
     return render_template(
         'index.html', 
-        todo_items = sorted_items,
+        todo_items = todo_items,
         num_todos = num_todos,
         num_complete_todos = num_complete_todos
     )
 
 @app.route('/todo/add', methods=['POST'])
 def add_todo():
-    new_todo = request.form.get('new-todo-item')
-    if new_todo.strip() != '':
-        add_item(new_todo)
+    name = request.form.get('todo-name')
+    description = request.form.get('todo-description')
+    date = request.form.get('todo-due-date')
+
+    if name.strip() != '':
+
+        to_add = {
+            'name': name,
+        }
+
+        if description.strip() != '':
+            to_add['desc'] = description
+        
+        if date.strip() != '':
+            to_add['due'] = datetime.strptime(date, '%d/%m/%Y').isoformat()
+        
+
+        trello = Trello()
+        trello.add_item(to_add)
     return redirect('/')
 
 @app.route('/todo/change-status/<id>', methods=['POST'])
 def change_todo_status(id):
-    todo_item = get_item(id)
-    current_status = todo_item['status']
-    new_status = 'Completed' if current_status == 'Not Started' else 'Not Started'
-    todo_item['status'] = new_status
+    credentials = get_trello_credentials()
+    trello = Trello()
+    todo_item = trello.get_item(id)
+    current_list = todo_item['idList']
 
-    save_item(todo_item)
+    new_list = None
 
+    if current_list == credentials['todo_list']:
+        new_list = credentials['completed_list']
+    else:
+        new_list = credentials['todo_list']
```

Comment:
Elsewhere you've managed to get Trello to handle these credentials which is quite nice from an encapsulation perspective. That is, `app.py` hasn't had to worry about this level of detail which can make it easier/safer to make changes later (and spoiler, in a few months time we'll make you swap out Trello! 😄 )

Could this be something that we capture more neatly inside the `update_item` function?

Similarly you get the current list with an extra API call to Trello, but could that be something that gets passed through as a hidden input from the form?